src/sokoban.cpp

```diff
@@ -109,16 +111,114 @@ bool Sokoban::make_move(Direction direction) {
             return true;
         }
     }
-
     return false;
 }
 
 bool Sokoban::move(Direction direction) {
     undone.clear();
-
     return make_move(direction);
 }
 
+bool Sokoban::move(unsigned int y, unsigned int x) {
+    auto origin = std::make_pair(py, px);
+    auto destination = std::make_pair(y, x);
+
+    // If the specified destination is the same, don't do anything
+    if (origin == destination) {
+        return false;
+    }
+
+    // If the specified destination is next to the player,
+    // try to move to that destination
+    
+    for (const auto &[dir, offset] : dir_offsets) {
+        auto delta = std::make_pair(origin.first + offset.first, origin.second + offset.second);
+        if (delta == destination) {
+            return move(dir);
+        }
+    }
+
+    std::queue<std::pair<unsigned int, unsigned int>> queue;
+    std::unordered_map<std::pair<unsigned int, unsigned int>, 
+        std::pair<unsigned int, unsigned int>, PairHash, PairEqual> visited;
+    bool destination_found = (py == y && px == x);
```

Comment:
I'm not sure if this boolean flag is needed. Can we call the "rebuild the path" logic from the `current == destination`? Maybe make it a separate func.